### PR TITLE
made a sorting by weight

### PR DESCRIPTION
**Pull request by End-rey**
Sorted the tags by the number of projects they are involved in

### CODE REVIEW

File: src/main/java/org/spburegistry/backend/controller/TagController.java [hunk 1]
```diff
@@ -23,6 +23,11 @@ public Iterable<TagTO> getAllTags() {
         return tagService.findAll();
     }
 
+    @GetMapping("/sortedByWeight")
+    public Iterable<TagTO> getAllTagsSortedByWeight() {
+        return tagService.findAllTagsSortedByWeight();
+    }
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/controller/TagController.java:29]: I think it should be implemented in 1 complex request like '/tags?sortedByWeight=true&filterSubstring="sbstr"'

File: src/main/java/org/spburegistry/backend/service/implementation/TagServiceImpl.java [hunk 2]
```diff
@@ -55,4 +55,12 @@ public TagTO addTag(TagTO tagTO) {
         return ConvertToTO.tagToTO(newTag);
 
     }
+
+    @Override
+    public Iterable<TagTO> findAllTagsSortedByWeight() {
+        return tagRepo.findAll().stream()
```
Comment (SerjVankovich) [inline src/main/java/org/spburegistry/backend/service/implementation/TagServiceImpl.java:61]: It is better to make sort on db query

**Review by SerjVankovich — COMMENTED** [review]
Fix comments

File: src/main/java/org/spburegistry/backend/service/implementation/TagServiceImpl.java (hunk 2 above)
Comment (End-rey) [reply src/main/java/org/spburegistry/backend/service/implementation/TagServiceImpl.java:61]: It's very difficult

File: src/main/java/org/spburegistry/backend/controller/TagController.java (hunk 1 above)
Comment (End-rey) [reply src/main/java/org/spburegistry/backend/controller/TagController.java:28]: Done

**Review by SerjVankovich — APPROVED** [review]
Very good!

**Pull request merged**